sklearn/impute/_iterative.py

```diff
@@ -627,7 +627,7 @@ def _initial_imputation(self, X, in_fit=False):
                 strategy=self.initial_strategy,
                 fill_value=self.fill_value,
                 keep_empty_features=self.keep_empty_features,
-            )
+            ).set_output(transform="default")
```

Comment:
I don't love this, cause it seems it could be quite buggy, as in, in our codebase I don't know where else we have such instances where we're not explicitly setting the output type.

I would suggest having a default which is to raise and be like: you haven't explicitly set this value for this estimator, please do so, and have that mode for scikit-learn at least.

This way we'd know we should always set it explicitly to: default, pandas, or the global config.

WDYT?

Reply:
I can get behind a new "raise_if_not_configured" configuration flag. Implementation-wise we'll need a standard place to check that configuration at `fit`. That can be in `_validate_data` or `_fit_context`. `_fit_context` is a more natural place but it is not used everywhere yet.

Reply:
> _fit_context is a more natural place but it is not used everywhere yet.

It will be the case when https://github.com/scikit-learn/scikit-learn/pull/26473 is merged